folly/experimental/coro/Traits.h

```diff
@@ -152,8 +152,8 @@ template <
             folly::Negation<detail::_has_member_operator_co_await<Awaitable>>>::
             value,
         int> = 0>
-Awaitable& get_awaiter(Awaitable&& awaitable) {
-  return awaitable;
+Awaitable&& get_awaiter(Awaitable&& awaitable) {
+  return std::forward<Awaitable>(awaitable);
 }
```

Comment:
I don't know anything about how this code is used, but it's a little scary that line 191 of this file has
```
template <typename Awaitable>
struct awaiter_type<Awaitable, std::enable_if_t<is_awaitable_v<Awaitable>>> {
  using type = decltype(get_awaiter(std::declval<Awaitable>()));
};
```
So IIUC, you're changing the meaning of `awaiter_type<A>::type` from an lvalue reference to an rvalue reference. Is that part of your intent?

If your goal is simply to fix the compiler warning and make this code C++23-compatible ("Simpler Implicit Move") without changing anything about its current behavior at all, then the correct fix is:
```
Awaitable& get_awaiter(Awaitable&& awaitable) {
  return static_cast<Awaitable&>(awaitable);
}
```

Reply:
I agree with you.

Reply:
```
template <
    typename Awaitable,
    std::enable_if_t<
        folly::Conjunction<
            !std::is_rvalue_reference<Awaitable&&>::value,
            is_awaiter<Awaitable>,
            folly::Negation<detail::_has_free_operator_co_await<Awaitable>>,
            folly::Negation<detail::_has_member_operator_co_await<Awaitable>>>::
            value,
        int> = 0>
Awaitable& get_awaiter(Awaitable&& awaitable) {
  return static_cast<Awaitable&>(awaitable);
}
```

I added a type check on the template function to prevent it from taking an rvalue-reference object.